tests/testthat/test-recipe.R

```diff
@@ -0,0 +1,70 @@
+skip_if_not_installed("recipes")
+skip_if_not_installed("plumber")
+
+library(plumber)
+library(recipes)
+
+trained_rec <-
+    recipe(mpg ~ disp + wt, mtcars) %>%
+    step_ns(wt) %>%
+    prep(retain = FALSE)
+
+v <- vetiver_model(trained_rec, "car-splines", prototype_data = mtcars[c("disp", "wt")])
```

Comment:
Notice that we are requiring the user to pass in some `prototype_data` (check out the `vetiver_ptype.recipe` method). This is what we have to do for ranger because the info on the training data isn't in there anywhere. If I was understanding Max correctly, this is what he was recommending. 

I want to note, though, that the original column names and types _are_ stored in a list, at `trained_rec$var_info`. Would there be a way to reconstruct the needed info (i.e. a `ptype`)?

Reply:
As it stands right now, there isn't a foolproof way of going from `trained_rec$var_info` to `ptype`s, since there is no guarantee that a 1-1 mapping can be found. This is much clearly seen since the type will be listed as `other` for any classes we don't currently specify.

I do however wish that this information was in recipes, as it is useful, even if we don't force the input checking. I will note and see if we can add such information in a future version. Which is another thing. The variable checking in recipes is done on a optional per-step basis, and can at times be quite loose. many steps doesn't care if input is double or integer. `step_dummy()` as a gross outlier doesn't do any type checking